src/main/java/com/marklogic/kafka/connect/source/MarkLogicSourceConfig.java

```diff
@@ -0,0 +1,87 @@
+package com.marklogic.kafka.connect.source;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+
+import java.util.Map;
+
+/**
+ * Defines configuration properties for the MarkLogic source connector.
+ */
+public class MarkLogicSourceConfig extends AbstractConfig {
+
+    public static final String CONNECTION_HOST = "ml.connection.host";
+    public static final String CONNECTION_PORT = "ml.connection.port";
+    public static final String CONNECTION_DATABASE = "ml.connection.database";
+    public static final String CONNECTION_SECURITY_CONTEXT_TYPE = "ml.connection.securityContextType";
+    public static final String CONNECTION_USERNAME = "ml.connection.username";
+    public static final String CONNECTION_PASSWORD = "ml.connection.password";
+    public static final String CONNECTION_TYPE = "ml.connection.type";
+    public static final String CONNECTION_SIMPLE_SSL = "ml.connection.simpleSsl";
+    public static final String CONNECTION_CERT_FILE = "ml.connection.certFile";
+    public static final String CONNECTION_CERT_PASSWORD = "ml.connection.certPassword";
+    public static final String CONNECTION_EXTERNAL_NAME = "ml.connection.externalName";
+    public static final String ENABLE_CUSTOM_SSL = "ml.connection.enableCustomSsl";
+    public static final String TLS_VERSION = "ml.connection.customSsl.tlsVersion";
+    public static final String SSL_HOST_VERIFIER = "ml.connection.customSsl.hostNameVerifier";
+    public static final String SSL_MUTUAL_AUTH = "ml.connection.customSsl.mutualAuth";
+
+    public static final String DMSDK_BATCH_SIZE = "ml.dmsdk.batchSize";
```

Comment:
My thought in the story for this was to do `ml.source.optic.batchSize` and `ml.source.optic.threadCount` so that all properties pertaining to reading rows via optic are prefixed under `ml.source.optic`. We could reuse `ml.dmsdk.batchSize`, since RowBatcher is part of DMSDK, but I was thinking the benefit of all the source-specific properties having the same prefix outweighs the benefit of reusing the ml.dmsdk properties. Thoughts?

Reply:
> We could reuse ml.dmsdk.batchSize, since RowBatcher is part of DMSDK
I don't particularly like the idea of prefixing all source-specific properties with ml.source.optic.

If you want them all (currently just 2) to start the same way, I'd go with ml.source.*